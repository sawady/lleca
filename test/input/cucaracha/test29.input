

fun p(x : Bool) {
    if x {
        putChar(84)
        putChar(114)
        putChar(117)
        putChar(101)
    } else {
        putChar(70)
        putChar(97)
        putChar(108)
        putChar(115)
        putChar(101)
    }
}

fun sep() {
    putChar(110)
    putChar(111)
    putChar(116)
    putChar(32)
}

fun eq() {
    putChar(32)
    putChar(61)
    putChar(32)
}

fun enter() {
    putChar(10)
}

fun fila(x : Bool) {
    r := not x
    sep() p(x) eq() p(r) enter()
}

fun main() {
  fila(False)
  fila(True)
}
