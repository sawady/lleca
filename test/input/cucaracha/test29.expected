Cons(
  Function(
    p,
    Cons(
      Parameter(
        x,
        Bool
      ),
      Nil
    ),
    Unit,
    Cons(
      StmtIf(
        ExprVar(
          x
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                84
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  114
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprConstNum(
                    117
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  putChar,
                  Cons(
                    ExprConstNum(
                      101
                    ),
                    Nil
                  )
                ),
                Nil
              )
            )
          )
        ),
        Just(
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  70
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprConstNum(
                    97
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  putChar,
                  Cons(
                    ExprConstNum(
                      108
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtCall(
                    putChar,
                    Cons(
                      ExprConstNum(
                        115
                      ),
                      Nil
                    )
                  ),
                  Cons(
                    StmtCall(
                      putChar,
                      Cons(
                        ExprConstNum(
                          101
                        ),
                        Nil
                      )
                    ),
                    Nil
                  )
                )
              )
            )
          )
        )
      ),
      Nil
    )
  ),
  Cons(
    Function(
      sep,
      Nil,
      Unit,
      Cons(
        StmtCall(
          putChar,
          Cons(
            ExprConstNum(
              110
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                111
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  116
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprConstNum(
                    32
                  ),
                  Nil
                )
              ),
              Nil
            )
          )
        )
      )
    ),
    Cons(
      Function(
        eq,
        Nil,
        Unit,
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                32
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  61
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprConstNum(
                    32
                  ),
                  Nil
                )
              ),
              Nil
            )
          )
        )
      ),
      Cons(
        Function(
          enter,
          Nil,
          Unit,
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  10
                ),
                Nil
              )
            ),
            Nil
          )
        ),
        Cons(
          Function(
            fila,
            Cons(
              Parameter(
                x,
                Bool
              ),
              Nil
            ),
            Unit,
            Cons(
              StmtAssign(
                r,
                ExprNot(
                  ExprVar(
                    x
                  )
                )
              ),
              Cons(
                StmtCall(
                  sep,
                  Nil
                ),
                Cons(
                  StmtCall(
                    p,
                    Cons(
                      ExprVar(
                        x
                      ),
                      Nil
                    )
                  ),
                  Cons(
                    StmtCall(
                      eq,
                      Nil
                    ),
                    Cons(
                      StmtCall(
                        p,
                        Cons(
                          ExprVar(
                            r
                          ),
                          Nil
                        )
                      ),
                      Cons(
                        StmtCall(
                          enter,
                          Nil
                        ),
                        Nil
                      )
                    )
                  )
                )
              )
            )
          ),
          Cons(
            Function(
              main,
              Nil,
              Unit,
              Cons(
                StmtCall(
                  fila,
                  Cons(
                    ExprConstBool(
                      False
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtCall(
                    fila,
                    Cons(
                      ExprConstBool(
                        True
                      ),
                      Nil
                    )
                  ),
                  Nil
                )
              )
            ),
            Nil
          )
        )
      )
    )
  )
)
